agent/agent.go

```diff
@@ -14,18 +14,25 @@ import (
 	"github.com/influxdata/telegraf/config"
 	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/models"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers/influx"
 )
 
 // Agent runs a set of plugins.
 type Agent struct {
-	Config *config.Config
+	Config 			*config.Config
+	iu     			*inputUnit
+	wg     			sync.WaitGroup
+	inputStatus 	map[string](chan string)
 }
 
 // NewAgent returns an Agent for the given Config.
 func NewAgent(config *config.Config) (*Agent, error) {
 	a := &Agent{
 		Config: config,
+		inputStatus : map[string](chan string) {},
```

Comment:
Unless you're explicitly defining starting values for a map, it's better practice to use the ```make``` keyword to initialize it.
eg: ```make(map[string]chan string)```